test/lit.site.cfg.py.in

```diff
@@ -36,6 +36,7 @@ config.circt_obj_root = "@CIRCT_BINARY_DIR@"
 config.circt_tools_dir = "@CIRCT_TOOLS_DIR@"
 config.verilator_path = "@VERILATOR_PATH@"
 config.esi_capnp = "@ESI_CAPNP@"
+config.bindings_python_enabled = "@CIRCT_BINDINGS_PYTHON_ENABLED@" == "ON"
```

Comment:
Upstream, we have this in the adjacent CMake file:

```
llvm_canonicalize_cmake_booleans(
  MLIR_BINDINGS_PYTHON_ENABLED
  LLVM_BUILD_EXAMPLES
  MLIR_CUDA_CONVERSIONS_ENABLED
  MLIR_CUDA_RUNNER_ENABLED
  MLIR_ROCM_CONVERSIONS_ENABLED
  MLIR_ROCM_RUNNER_ENABLED
  MLIR_SPIRV_CPU_RUNNER_ENABLED
  MLIR_VULKAN_RUNNER_ENABLED
  )
```

Then this in the site cfg:

```
config.enable_bindings_python = @MLIR_BINDINGS_PYTHON_ENABLED@
```

A bit more robust.